### PR TITLE
Sitemaps: add `lastmod` for individual posts and the homepage

### PR DESCRIPTION
**Pull request by swissspidy**
This resolves the most critical aspect of that Trac ticket for "Last modified".

What it does:

* Adds `lastmod` to all individual post objects (of any post type) in the sitemap.
	* This is trivial to add because this information is directly available.
* Adds `lastmod` to the homepage sitemap entry if the homepage is set to display the latest posts.
	* This is an approximation that works by fetching the most recent posts (based on their _publish date_) and then sorting them in PHP by their _modified date_.

As [per my comment](https://core.trac.wordpress.org/ticket/52099#comment:22), the latter is mostly nice to have but not critical. If we think this approximation is not good enough, we can omit it for now.

What it does not:

* No `lastmod` is added for the individual sitemap pages in the sitemap index
* No `lastmod` is added for term archives or user archives

Those are difficult to add without any proper mechanism to determine and cache last modified date. As [per my comment](https://core.trac.wordpress.org/ticket/52099#comment:22), this should not be an issue.

Trac ticket: https://core.trac.wordpress.org/ticket/52099

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by swissspidy** [comment]
@pbiron Would love to get your review on this one

**Comment by swissspidy** [comment]
@joemcgill Since you were involved with sitemaps originally, I would appreciate your eyes on this PR.

tl;dr is that lastmod is useful for search engines, so when we can easily expose it (such as for posts and the homepage), we should, but it's also no big deal if it's missing (e.g. for term archives)

**Review by joemcgill — APPROVED** [review]
This looks good to me.

It's been several years since I worked on the original proof of concept for this feature, but if memory serves, we initially included `lastmod` values for these types of pages and ended up removing them for consistency, since some values are difficult to calculate in a performant way. At the time, the guidance we were getting was that these values were not being used as a significant signal by search engines, and extenders were able to filter these values to add `lastmod` if needed. [Here's the PR](https://github.com/GoogleChromeLabs/wp-sitemaps/pull/145) where the initial implementation was removed.

It's possible that some of those original ideas could be used to further extend the places where core makes lastmod available.

**Comment by swissspidy** [comment]
Thanks @joemcgill! That's how I remember it too. Now the guidance has slightly changed, and providing lastmod where easily possible is now preferrable.

This PR will close the existing ticket, but a new one can be opened for lastmod elsewhere and whatever is required for that.

**Comment by swissspidy** [comment]
Committed in https://core.trac.wordpress.org/changeset/56985
